src/app/give-help/accommodation-form/accommodation-form.component.ts

```diff
@@ -53,23 +51,20 @@ export class AccommodationFormComponent implements OnInit {
     }
   }
 
-  findPrefix(phoneCountryCode: string) {
-    this.phonePrefix = PREFIXES.find((v) => v.prefix === phoneCountryCode)?.prefix || '';
+  phoneNumberChange(phone: PhoneNumber) {
+    this.phone = phone;
   }
 
-  onPhoneNumberChange($event: Event) {
-    let val = ($event.target as HTMLInputElement).value;
-    val = val.replace(MATCH_NON_DIGITS, '').replace(MATCH_SPACES, '');
-    this.phoneInput.nativeElement.value = val;
-    this.phoneNumber = val;
+  findPrefix(prefix: string) {
+    return PREFIXES.find((v) => v.prefix === prefix)?.prefix || '';
   }
 
   preparePhoneNumber() {
-    this.data.phoneNumber = this.phonePrefix + this.phoneNumber;
+    this.data.phoneNumber = this.phone.prefix + this.phone.phoneNumber;
```

Comment:
`${this.phone.prefix}${this.phone.phoneNumber}`

Reply:
To jest jakiś goodpractice? Jest czymś poparte czy poprostu Twoje preferencje?

Reply:
Nie zaleca się używania operatora `+` ponieważ wykonuje on rzutowania w zależności od typów po lewej i prawej stronie operatora. Tym bardziej, że w naszym przypadku nazwy propsow sugerują liczby.
W tym przypadku może nawet `.concat()` byłby bardziej właściwy niż string literał.

Reply:
done